runtime/src/main/java/io/quarkus/vault/runtime/config/VaultBootstrapConfig.java

```diff
@@ -250,66 +249,76 @@ public class VaultBootstrapConfig {
      *
      * @asciidoclet
      */
-    @ConfigItem
-    public Map<String, CredentialsProviderConfig> credentialsProvider;
+    Map<String, CredentialsProviderConfig> credentialsProvider();
 
     /**
      * Transit Engine
      */
-    @ConfigItem
     @ConfigDocSection
-    public VaultTransitConfig transit;
+    VaultTransitConfig transit();
+
+    /**
+     * Deprecated.
+     */
+    @Deprecated
+    @WithName("devservices")
+    Map<String, String> devServices();
 
-    public VaultAuthenticationType getAuthenticationType() {
-        if (authentication.kubernetes.role.isPresent()) {
+    /**
+     * Deprecated.
+     */
+    @Deprecated
+    Map<String, String> health();
+
+    default VaultAuthenticationType getAuthenticationType() {
+        if (authentication().kubernetes().role().isPresent()) {
             return KUBERNETES;
-        } else if (authentication.isUserpass()) {
+        } else if (authentication().isUserpass()) {
             return USERPASS;
-        } else if (authentication.isAppRole()) {
+        } else if (authentication().isAppRole()) {
             return APPROLE;
         } else {
             return null;
         }
     }
 
-    @Override
-    public String toString() {
+    default String toString(LogConfidentialityLevel level) {
```

Comment:
rename to `toSecuredString()` with no argument
hardcode the level for each attribute, according to their own sensitivity